common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -67,79 +111,756 @@ public void onClose() {
+                .size((int) (dividedWidth * 8), 20)
+                .build();
+
+        deleteSelectedButton.active = false;
+
+        this.addRenderableWidget(deleteSelectedButton);
+        // endregion
+
+        // region select buttons
+        deselectAllButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.poiManagementGui.deselectAll"),
+                        (button) -> toggleSelectAll(false))
+                .pos((int) dividedWidth, (int) (dividedHeight * 58))
+                .size((int) (dividedWidth * 8), 20)
+                .build();
+
+        deselectAllButton.active = false;
+
+        this.addRenderableWidget(deselectAllButton);
+
+        selectAllButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.poiManagementGui.selectAll"),
+                        (button) -> toggleSelectAll(true))
+                .pos((int) dividedWidth, (int) (dividedHeight * 58) - 25)
+                .size((int) (dividedWidth * 8), 20)
+                .build();
+
+        this.addRenderableWidget(selectAllButton);
+        // endregion
+
+        // region up/down buttons
+        upButton = new Button.Builder(Component.literal("ʌ"), (button) -> updateSelectedPoiPositions(-1))
+                .pos((width / 2) - 22, (int) (dividedHeight * 58))
+                .size(20, 20)
+                .build();
+
+        upButton.visible = false;
+
+        this.addRenderableWidget(upButton);
+
+        downButton = new Button.Builder(Component.literal("v"), (button) -> updateSelectedPoiPositions(1))
+                .pos((width / 2) + 2, (int) (dividedHeight * 58))
+                .size(20, 20)
+                .build();
+
+        downButton.visible = false;
+
+        this.addRenderableWidget(downButton);
+        // endregion
+
+        // region icon filter button
+        int filterButtonWidth = (int) (dividedWidth * 10);
+
+        filterButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.poiManagementGui.filter"), (button) -> {
+                            scrollOffset = 0;
+                            McUtils.mc().setScreen(IconFilterScreen.create(this, filteredIcons));
+                        })
+                .pos((int) (dividedWidth * 44), (int) (dividedHeight * 3))
+                .size(filterButtonWidth, 20)
+                .build();
+
+        this.addRenderableWidget(filterButton);
+        // endregion
+
+        // region search bar
+        searchInput = new TextInputBoxWidget(
+                (int) (dividedWidth * 10) + 5,
+                (int) (dividedHeight * 3),
+                (int) (backgroundWidth - filterButtonWidth - 10),
+                20,
+                (s) -> {
+                    scrollOffset = 0;
+                    populatePois();
+                },
+                this,
+                searchInput);
+
+        this.addRenderableWidget(searchInput);
```

Comment:
I think you forgot to copy the old searchInput's value on reinit, if it exists already

Reply:
It works? Isn't that what passing searchInput into the constructor does?

```
if (oldWidget != null) {
    this.textBoxInput = oldWidget.textBoxInput;
    setCursorAndHighlightPositions(oldWidget.cursorPosition);
    this.renderColor = oldWidget.renderColor;
}
```

Reply:
Yeah I missed that. Some other search box didn't support that, not sure where I saw it.